File: Classwork/Java/Marker.java


package com.github.ndbellew.marker;
/*
*
*  @Author ndbellew
*/
public class Marker{
  private Color color;
  private boolean capped;

  Marker(Color color){
    this.color = color;
    this.capped = true;
  }

  public Color getColor() {
    return color;
  }
  public void setColor(Color color){
    this.color = color;
  }
  public boolean isCapped(){
    return capped;
  }
  public void setCapped(boolean capped){
    this.capped = capped;
  }
  public void draw() throws BadMarkerState{
    if(!isCapped()) {
      System.out.println("drawing in "+getColor());
    }else{
      throw Marker.BAD_MARKER_CAPPED_DRAW;
    }
  }
  static public BadMarkerState BAD_MARKER_CAPPED_DRAW =
    new BadMarkerState("Drawing with capped marker is bad");
}

File: Classwork/CPlusPlus/marker.cpp

#include "Marker.h"

namespace marker {
    Marker::Marker(Color _color) {
      color = _color;
      capped = true;
    }
   Marker::Color getColor() {
    return this->color;
  }

  Marker::bool isCapped(){
    return this->capped;
  }

  Marker::void setColor(Color _color){
    this->color = _color;
  }

  Marker::void setCapped(_capped){
    this->capped=_capped;
  }
  Marker::void draw(){
    if(!isCapped()){
      cout<<"drawing in  "<<getColor()<<endl;
    }else{
      throw BAD_MARKER_CAPPED_DRAW
    }
  }
}
